src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js

```diff
@@ -7,10 +7,34 @@ export default {
                 },
 
                 toObject(message, options) {
-                    const seconds = Number(message.seconds) * 1000;
-                    const date = new Date(seconds);
+                    const { seconds, nanos } = message;
+                    const MILIS_PER_SECOND = 1000;
+                    const NANOS_PER_MILLISECOND = 1000000;
+                    const NANOS_PER_MICROSECOND = 1000;
 
-                    return date.toJSON();
+                    const date = new Date(Number(seconds) * MILIS_PER_SECOND);
+
+                    let result = date.toJSON();
+
+                    // Support for Nano precision
+                    if (nanos !== 0) {

```

Comment:
Nice find. Thanks!

Could you add some test cases to serializer-protobuf-spec.js, with mocked nanoseconds in proto-price.js?

Reply:
Yes, will do.